### DISCUSSION TITLE
How do you add Songs, Characters, Stages and Weeks to 1.7?

### DISCUSSION THREAD

**Discussion by BrandonHatesAnime** [KadeArchive/Kade-Engine#2434] [Q&A] [answered]
Hi, Kade Engine has really changed a lot compared to previous versions.
I would like to review, take note and verify how to add songs, characters, stagers and weeks to the current Kade Engine (1.7).
Can someone please be nice enough to specifically tell me a detailed tutorial on how to do that? Thanks.

**Comment by isakube** [comment, marked as answer]
- stages are weird as shit now and i haven't bothered to learn them
- songs are literally the exact same i think
- characters are ALSO the exact same, but thanks to MasterEric (#2277) you can also make a json sorta like Modding+. someone else probably knows how that works.
- weeks are also the same. `weekData` is a function now tho for some reason

**Reply by BrandonHatesAnime** [reply]
You add stages like same but in stage.hx instead of Playstate.hx
